mgmtsystem/virtualcenter.py

```diff
@@ -512,23 +594,23 @@ def clone_vm(self, source, destination, resourcepool=None, datastore=None, power
                  sparse=False, template=False, provision_timeout=1800, progress_callback=None,
                  allowed_datastores=None, cpu=None, ram=None, **kwargs):
         try:
-            if mobs.VirtualMachine.get(self.api, name=destination).name == destination:
+            if self._get_obj(vim.VirtualMachine, name=destination).name == destination:
                 raise Exception("VM already present!")
-        except ObjectNotFoundError:
+        except VMInstanceNotFound:
             pass
 
         if progress_callback is None:
             progress_callback = partial(self._progress_log_callback, self.logger,
                 source, destination)
 
-        source_template = mobs.VirtualMachine.get(self.api, name=source)
+        source_template = self._get_obj(vim.VirtualMachine, name=source)
 
-        vm_clone_spec = self.api.create("VirtualMachineCloneSpec")
-        vm_reloc_spec = self.api.create("VirtualMachineRelocateSpec")
+        vm_clone_spec = vim.VirtualMachineCloneSpec()
+        vm_reloc_spec = vim.VirtualMachineRelocateSpec()
         # DATASTORE
         if isinstance(datastore, basestring):
```

Comment:
as this is freestanding code outside of the changed code i would consider a followup acceptable as well